cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/distributed/LogLevelUpdater.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.app.runtime.distributed;
+
+import org.apache.twill.api.logging.LogEntry;
+
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Provides methods to update or reset log levels of a program at runtime in distributed mode.
+ */
+public interface LogLevelUpdater {
+
+  /**
+   * Update the log levels of the program.
+   *
+   * @param runnableName The name of the runnable to update the log level. If {@code null}, all runnables will get
+   *                     updated.
+   * @param logLevels The {@link Map} contains the requested logger name and log level.
+   */
+  void updateLogLevels(@Nullable String runnableName, Map<String, LogEntry.Level> logLevels) throws Exception;
```

Comment:
cdap doesn't really have the concept of `runnable`. Shouldn't it be component name? Also, it's better to have `Nullable` parameter locates at the end of the parameter list.